cloudflare/data_source_waf_rules_test.go

```diff
@@ -52,6 +52,27 @@ func TestAccCloudflareWAFRules_MatchDescription(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareWAFRules_DefaultMode(t *testing.T) {
+	t.Parallel()
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("data.cloudflare_waf_rules.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareWAFRulesConfig(zoneID, map[string]string{}, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWAFRulesDataSourceID(name),
+					resource.TestCheckResourceAttrSet(name, "rules.#"),
```

Comment:
i'm afraid this assertion doesn't do assert anything useful for us here. this test is identical to some others and doesn't actually cover the functionality you've added. 

perhaps have a look at `testAccCheckCloudflareWAFRulesDataSourceID` and i suspect you'll need to programmatically loop through the resource rules and confirm the key is present in the state following this change.